### PR TITLE
Bootstrap project command - phase II

### PR DESCRIPTION
**Pull request by AronNovak**
#417

### CODE REVIEW

**Comment by AronNovak** [comment]
image

**Comment by AronNovak** [comment]
image

**Comment by AronNovak** [comment]
Now it works correctly to
 - prepare GH repo
 - create new Pantheon project
 - prepare Pantheon project
 - create QA env
 - install environments
 - lock environments

What still needs to be done:
 - integrate autodeploy configuration, so it does not have to be a different command invocation after project setup.

File: robo-components/BootstrapTrait.php [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+    $this->verifyRequirements($project_name, $organization, $project_machine_name, $terminus_token, $github_token, $docker_mirror_url, $http_basic_auth_user, $http_basic_auth_password);
+
+    $this->prepareGithubRepository($project_name, $organization, $project_machine_name, $github_repository_url, $docker_mirror_url);
+
+    $this->createPantheonProject($terminus_token, $project_name, $project_machine_name, $organization);
+
+    $this->deployPantheonInstallEnv('dev', $project_machine_name);
+    $this->deployPantheonInstallEnv('qa', $project_machine_name);
+
+    if ($http_basic_auth_user && $http_basic_auth_password) {
```
Comment (amitaibu) [inline robo-components/BootstrapTrait.php:46]: Let's move this check into `lockPantheonEnvironments`

File: robo-components/BootstrapTrait.php [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+      $this->lockPantheonEnvironments($project_machine_name, $http_basic_auth_user, $http_basic_auth_password);
+    }
+    else {
+      $this->say("No HTTP basic auth credentials were provided. Pantheon environments will not be locked.");
+    }
+
+    $this->say("Bootstrap completed successfully.");
+    $this->say("You might want to run the following commands to properly place the project:");
+    $this->say("mv .bootstrap ../$project_machine_name");
```
Comment (amitaibu) [inline robo-components/BootstrapTrait.php:55]: What's inside `.bootstrap`?

Also, maybe we should do it for them?

**Review by amitaibu — COMMENTED** [review]
Great stuff, one comment

File: robo-components/BootstrapTrait.php (hunk 2 above)
Comment (AronNovak) [reply robo-components/BootstrapTrait.php:55]: We are bound to what's mounted into DDEV, that's why we cannot easily move it, but i will think about it, also we have an issue related to invoke the autodeploy config, what would be executed in another DDEV project, that needs to be refactored to be able to do it in one step.

Comment (AronNovak) [reply robo-components/BootstrapTrait.php:55]: Inside it's the new project's GH repository.

File: pantheon_template/settings.pantheon.php [hunk 3]
```diff
@@ -0,0 +1,204 @@
+<?php
```
Comment (mariano-dagostino) [inline pantheon_template/settings.pantheon.php:1]: @AronNovak Should we download this file directly from https://github.com/pantheon-systems/drops-8/blob/495b94f873878ee0b06651307059eccab337562d/sites/default/settings.pantheon.php

**Review by mariano-dagostino — APPROVED** [review]

File: pantheon_template/settings.pantheon.php (hunk 3 above)
Comment (AronNovak) [reply pantheon_template/settings.pantheon.php:1]: @mariano-dagostino Good point. Do you have a suggestion on which branch to use? 
https://github.com/pantheon-systems/drops-8/tree/default-9.x/sites is not the same that you linked for instance, it seems to be a bit chaotic.

Comment (amitaibu) [reply pantheon_template/settings.pantheon.php:1]: I don't think it will change much, so we can start with the file we have

**Comment by amitaibu** [comment]
Thanks

**Pull request merged**